doc/source/ray-air/doc_code/tf_starter.py

```diff
@@ -54,16 +54,17 @@ def train_func(config: dict):
 
     results = []
     for _ in range(epochs):
-        tf_dataset = prepare_dataset_shard(
-            dataset.to_tf(
-                label_column="y",
-                output_signature=(
-                    tf.TensorSpec(shape=(None, 1), dtype=tf.float32),
-                    tf.TensorSpec(shape=(None), dtype=tf.float32),
-                ),
-                batch_size=batch_size,
-            )
+        tf_dataset = tf.data.Dataset.from_generator(
+            lambda: (
+                (tf.expand_dims(batch["x"], 1), batch["y"])
```

Comment:
We need this dimension expansion in some examples (same for torch), but not in all. Can we get rid of this by adding something to the API?

Also c @c21 here's how to convert to_tf to the new API